include/arch/x86/linker.ld

```diff
@@ -180,10 +205,18 @@ SECTIONS
 	SECTION_PROLOGUE(_BSS_SECTION_NAME, (NOLOAD OPTIONAL),)
 	{
 	/*
+	 * Because of text section displacement to follow startup ones
+	 * on LMA, align back to page here
+	 */
+#ifdef CONFIG_JAILHOUSE
```

Comment:
I'm confused by this. Can you provide more detail?

Reply:
Maybe I have to rephrase the comment, help is appreciated. With this code, I get

```
In Page directory entry 0    : valid start address = 0x000f4000, end address = 0x003fffff
In Page directory entry 0    : valid start address = 0x00000000, end address = 0x000f3fff
```
as PDE entries for those ranges, without I get

`Memory Regions are not page aligned 0xf3c40`

instead.

Reply:
somehow we are passing an unaligned address/size to an MMU_BOOT_REGION() call. What MMU_BOOT_REGION call is generating this error? What about this patch shifts it?

Reply:
It seems to be `__kernel_ram_start`. I have no clue why originally it would end up page aligned by just reading that linker code. What we shift on purpose is the LMA of the text section, to be placed just after the (new) startup one. This is how Jailhouse works, by having the jump from the 16-bit (boot) section to the proper 32-bit text be a fall-through. For now I'll keep looking, any help is appreciated.

Reply:
It seems the page alignment comes for free from `PHYS_RAM_ADDR`, since it's the first section in that memory region. With this port, we lay out every section in a sequence and we may end up unaligned. Either we ease on the checking script or we keep this line. I'll pack all the patches up again as soon as I work with upstream Jailhouse to have the new, necessary patches on their "next" branch.